lib/streamlit/config.py

```diff
@@ -2046,3 +2103,6 @@ def func_with_lock() -> None:
 # may edit config options based on the values of other config options.
 on_config_parsed(_check_conflicts, lock=True)
 on_config_parsed(_set_development_mode)
+# Update server.trustedUserHeaders from any JSON string that was set. Take out the
+# lock, since this is mutating the config.
+on_config_parsed(_parse_trusted_user_headers, lock=True)
```

Comment:
We have at least one other config case where we support parsing json: fontFaces:

https://github.com/streamlit/streamlit/blob/ab2ec1c00a6d69c7b551086fb785774f3a01d657/lib/streamlit/config.py#L1215-L1244

In that case, we are applying the parsing at runtime:

https://github.com/streamlit/streamlit/blob/ab2ec1c00a6d69c7b551086fb785774f3a01d657/lib/streamlit/runtime/app_session.py#L984-L1011

This has the benefit that the developer can play around with the config during development without restarting Streamlit, but it comes with added overhead. We could also apply the `trustedUserHeaders` parsing in `browser_websocket_handler`  directly, but no strong opinion from me on this.